### PR TITLE
adding eventOverlay

### PR DESCRIPTION
**Pull request by matanagalili**

### CODE REVIEW

File: src/main/scala/com/supersonic/datadog/datadog_schemas.scala [hunk 1]
```diff
@@ -2,6 +2,7 @@ package com.supersonic.datadog
```
Comment (ncreep2) [inline src/main/scala/com/supersonic/datadog/datadog_schemas.scala:2]: Please format the file

File: src/main/scala/com/supersonic/datadog/datadog_schemas.scala [hunk 2]
```diff
@@ -21,7 +22,8 @@ case class TemplateVariable(name: String, prefix: String, default: Option[String
 object Graph {
   case class Definition(requests: List[Request],
                         visualization: Visualization,
-                        yAxis: Option[YAxis] = None)
+                        yAxis: Option[YAxis] = None,
+                        eventOverlay: EventOverlay)
```
Comment (ncreep2) [inline src/main/scala/com/supersonic/datadog/datadog_schemas.scala]: Please make this optional (with a default)

File: src/main/scala/com/supersonic/datadog/datadog_schemas.scala [hunk 3]
```diff
@@ -67,6 +69,24 @@ object Graph {
     case class Op(name: String)
   }
 
+  sealed trait EventOverlay {
+    def render :String = this match {
+      case EventTag(name, value, eventName) =>
+        val renderedValue = value.map(v => s":$v").getOrElse("")
+        s"$name$renderedValue + $eventName"
```
Comment (ncreep2) [inline src/main/scala/com/supersonic/datadog/datadog_schemas.scala]: Is the `+` here intentional?

File: src/main/scala/com/supersonic/datadog/datadog_schemas.scala [hunk 4]
```diff
@@ -67,6 +69,24 @@ object Graph {
...
+      case EventFromTemplateVariable(value, eventName) => "$" + value.name + eventName
```
Comment (ncreep2) [inline src/main/scala/com/supersonic/datadog/datadog_schemas.scala]: You can use string interpolation, here and below

File: src/main/scala/com/supersonic/datadog/datadog_schemas.scala (hunk 2 above)
Comment (matanagalili) [reply src/main/scala/com/supersonic/datadog/datadog_schemas.scala]: Done

File: src/main/scala/com/supersonic/datadog/datadog_schemas.scala (hunk 3 above)
Comment (matanagalili) [reply src/main/scala/com/supersonic/datadog/datadog_schemas.scala]: No, thanks

File: src/main/scala/com/supersonic/datadog/datadog_schemas.scala (hunk 4 above)
Comment (matanagalili) [reply src/main/scala/com/supersonic/datadog/datadog_schemas.scala]: Done

**Review by ncreep2 — APPROVED** [review]

**Pull request merged**